### PR TITLE
add pipelines

### PR DESCRIPTION
**Pull request by AleDore**
This PR add azure devops pipelines

### CODE REVIEW

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
@AleDore can I have a push permission on this repo? thanks

File: .devops/code-review-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+              echo "validating $apispec"
+              npx oval validate -p $apispec
+              result=$?
+              if [ ! $result = 0 ]; then
+                valid=0
+                echo "$apispec failed with code $result"
+              fi
+            done
+            [[ $valid = 1 ]] && exit 0 || exit 1
+          displayName: 'Validate API specifications'
```
[the remark this answers is not in the document]
Comment (AleDore) [reply .devops/code-review-pipelines.yml:68]: Nope i will remove it

File: .devops/code-review-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            yarn generate:proxy-models
+          displayName: 'Generate definitions'        
+        
+        - script: |
+            yarn generate:test-certs
+          displayName: 'Generate test certificates'        
+
+        - script: |
+            yarn test:coverage
+          displayName: 'Unit tests exec'
```
[the remark this answers is not in the document]
Comment (AleDore) [reply .devops/code-review-pipelines.yml:104]: done

**Comment by pasqualedevita** [comment]
list of changes https://github.com/pagopa/hub-spid-login-ms/pull/4/commits/9e4fd9b18a64d74def40171a8b8ed577d3db785e:
1. add `.node-version` `.nvmrc ` and removed NODE_VERSION variable
2. removed duplicated variables section
3. change vmImage to ubuntu-latest (app service plan is on linux)
4. added more variables for multi environment (prod, uat)
5. added new variable $(DEFAULT_BRANCH) instead of master/main (we need to add it into azdevops)
6. refactor deploy stage with slot (deploy on staging slot, healthcheck, swap production and staging slot)

PS: app service does not has staging slot, maybe we have to add it or add an optional deploy stage without slot

**Review by pasqualedevita — APPROVED** [review]
LGTM

**Pull request merged**